### ISSUE TITLE
[STORY: E001-S002] Convert the Markdown Issue Templates to Simplified YAML Issue Forms and Simplify the Pull Request Template

### ISSUE THREAD

**Issue by antonio-crispino** [antonio-crispino/.github#14]
# Story: [E001](https://github.com/antonio-crispino/.github/issues/1 "Parent Issue (Epic)")-S002 <span id="s-story">

**As a** project collaborator, **I want** to convert the existing Markdown issue templates into YAML issue forms and simplify the structure, without loss of information, of both those and the Markdown pull request template, **so that** the process of creating and managing issues and pull requests becomes simpler.

## Table of Contents <span id="s-table-of-contents">

1. [Description :memo:](#s-description "A detailed description of the Story. This includes any relevant context, requirements, and constraints.")
2. [Priority :exclamation:](#s-priority "The priority of the Story (MoSCoW method).")
3. [User Story Points (USP) :star:](#s-user-story-points-usp "The estimated effort required to complete the Story.")
4. [Dependencies :chains:](#s-dependencies "A list of dependencies required for the completion of the Story.")
5. [Acceptance Criteria :heavy_check_mark:](#s-acceptance-criteria "A list of conditions that must be met for the Story to be considered complete. This can use the Given-When-Then (GWT) format and other statement formats if appropriate.")
6. [Subtasks :clipboard:](#s-subtasks "A list of subtasks required to complete the Story. These do not need to have their own Subtask issues, but they could.")
7. [Parent Issue (Epic) :link:](#s-parent-issue-epic "The parent issue (Epic) for this Story, if it has one.")
8. [Resources :books:](#s-resources "A list of useful resources/links for the Story.")
9. [Notes :pencil2:](#s-notes "A list of additional information about the Story.")

## 1 - Description :memo: <span id="s-description">

> _A detailed description of the Story. This includes any relevant context, requirements, and constraints._

This Story involves converting the existing Markdown issue templates (Epic, Story, Task, Bug, and Subtask) to YAML issue forms and simplifying the structure (without loss of information) of both those and the Markdown pull request template.

Regarding the issue templates, the following sections will be removed should that issue template have them. Also, removing them will not cause any loss of information.

- **Secondary Title**: This is repetitive because there already is a main title for the issue.
- **Brief Summary**: Anything described here is already described in the main description section.
- **Table of Contents**: This takes up a significant amount of space on the page and is not really necessary. This is practical in longer documents.
- **Parent/Grandparent Issue Links**: This is repetitive because the title of the issue already has any parent or grandparent issue within its name. Note that sections such as related and child issues will be kept.
- **Resources**: This is an entire section simply for links. Links can be added at any place within the document. Therefore, removing this section would simplify the document without any loss of information.
- **Notes**: This is an entire section simply for extra notes. These notes can be added at any place within the document. Therefore, removing this section would simplify the document without any loss of information.

Some issue templates include extra sections that would be more suitable as labels, rather than requiring separate sections for input. Therefore, in the issue forms, these sections should be substituted with details about the corresponding label choices. These should then be erased after the issue has been submitted. These sections are "**Type**", "**Priority**", and "**Points**".

## 2 - Priority :exclamation: <span id="s-priority">

> _The priority of the Story (MoSCoW method). Select one of the following priorities:_
>
> 1. :sunny: `must-have`
> 2. :large_orange_diamond: `should-have`
> 3. :large_blue_diamond: `could-have`
> 4. :no_entry: `won't-have`

`should-have`

## 3 - User Story Points (USP) :star: <span id="s-user-story-points-usp">

> _The estimated effort required to complete the Story. This is estimated using user story points (USP). The number of user story points allocated is selected from the following Fibonacci-like sequence: [1, 2, 3, 5, 8, 13, 20, 40, 100]._

8 USP

## 4 - Dependencies :chains: <span id="s-dependencies">

> _A list of dependencies required for the completion of the Story._

1. **The Story** cannot start until Task [E001-T001](https://github.com/antonio-crispino/.github/issues/2) is completed.
2. **The Story** cannot be submitted until Story [E001-S001](https://github.com/antonio-crispino/.github/issues/3) is completed.

## 5 - Acceptance Criteria :heavy_check_mark: <span id="s-acceptance-criteria">

> _A list of conditions that must be met for the Story to be considered complete. This can use the Given-When-Then (GWT) format and other statement formats if appropriate._

1. **Given** the GitHub website is open, **when** a contributor creates a new issue, **then** there are Epic, Story, Task, Bug, and Subtask issue form options available and they have a simpler structure than the previous issue templates.
2. **Given** an issue is complete, **when** a PR is made, **then** a general PR template is present that has a simpler structure than the previous PR template.

## 6 - Subtasks :clipboard: <span id="s-subtasks">

> _A list of subtasks required to complete the Story. These do not need to have their own Subtask issues, but they could._

- [x] 1. Create YAML Epic issue form that also has a simpler structure (without loss of information) than the Markdown Epic issue template.
- [x] 2. Create YAML Story issue form that also has a simpler structure (without loss of information) than the Markdown Story issue template. In addition to the sections that should be removed [described above](#s-description), the user story statement should be given its own section.
- [x] 3. Create YAML Task issue form that also has a simpler structure (without loss of information) than the Markdown Task issue template.
- [x] 4. Create YAML Bug issue form that also has a simpler structure (without loss of information) than the Markdown Bug issue template.
- [x] 5. ~~Create YAML Subtask issue form that also has a simpler structure (without loss of information) than the Markdown Subtask issue template.~~
- [x] 6. Delete the Markdown issue templates. Be sure to adjust the `CONTRIBUTING.md` and `README.md` files accordingly.
- [x] 7. Simplify (without loss of information) the current Pull Request (PR) template. As with the issue templates, the secondary title, brief summary, table of contents, resources, and notes should be removed for the same reasons. The "Type" and "Related Issue(s)" section should remain commented out so that they are not part of the output. They should have their own section outputs because the type should be a label while the related issue(s) should be linked in the "Development" section of the pull request sidebar.
- [x] 8. Delete Subtask issue template/form altogether. This issue template/form should not be needed because Stories and Tasks are intended to encapsulate a manageable amount work that can be completed as-is, without the need for decomposition into smaller tasks (Subtasks). Should a Story or Task feel like too much work, it might be because it should be an Epic broken down into multiple different Stories and Tasks instead. Bugs should not be decomposed into Subtasks either, and if fixing the Bug requires too much work, maybe this should become an entire change to the software and structured differently in the hierarchy of issues.

## 7 - Parent Issue (Epic) :link: <span id="s-parent-issue-epic">

> _The parent issue (Epic) for this Story, if it has one._

[Epic E001](https://github.com/antonio-crispino/.github/issues/1 "Parent Issue (Epic)"): Create Default Community Health Files

## 8 - Resources :books: <span id="s-resources">

> _A list of useful resources/links for the Story._

None

## 9 - Notes :pencil2: <span id="s-notes">

> _A list of additional information about the Story._

None
